ts/nni_manager/training_service/local/localTrainingService.ts

```diff
@@ -253,7 +253,14 @@ class LocalTrainingService implements TrainingService {
 
             return Promise.resolve();
         }
-        tkill(trialJob.pid, 'SIGKILL');
+        tkill(trialJob.pid, 'SIGTERM');
+        const pid = trialJob.pid;
+        setTimeout((pid: number) => {
+            if (isAlive(pid)) {
```

Comment:
This if check is not atomic.
Please call tkill can catch exception.

Reply:
fixed it.